HttpUtils.cpp

```diff
@@ -194,31 +194,31 @@ HttpUtils::GetAll(BUrl url, BHttpHeaders* responseHeaders, bigtime_t timeOut,
 		return data;
 
 	HttpIOReader reader(data, responseHeaders, sizeLimit);
-	HttpRequest request(url, false, "HTTP", &reader);
+	BHttpRequest* request = dynamic_cast<BHttpRequest*>(BUrlProtocolRoster::MakeRequest(url.UrlString().String(), &reader, NULL));
```

Comment:
you are missing a "delete request;" at the end of the function to avoid a memory leak.